### PR TITLE
[web] Move `EmbeddingStrategy` and `DimensionsProvider` out of `FlutterViewEmbedder`

### PR DESCRIPTION
**Pull request by mdebbar**
Part of https://github.com/flutter/flutter/issues/134443
Part of https://github.com/flutter/flutter/issues/117098

### CODE REVIEW

**Comment by ditman** [comment]
(I think this PR Fixes, or is also part of: https://github.com/flutter/flutter/issues/117098)

File: lib/web_ui/lib/src/engine/window.dart [hunk 1]
```diff
@@ -40,22 +41,35 @@ const int kImplicitViewId = 0;
...
   EngineFlutterView._(
     this.viewId,
     this.platformDispatcher,
-  );
+    // This is nullable to accommodate the legacy `EngineFlutterWindow`. In
+    // multi-view mode, the host element is required for each view (as reflected
+    // by the public `EngineFlutterView` constructor).
+    DomElement? hostElement,
+  )   : embeddingStrategy = EmbeddingStrategy.create(hostElement: hostElement),
+        _dimensionsProvider = DimensionsProvider.create(hostElement: hostElement);
```
Comment (ditman) [inline lib/web_ui/lib/src/engine/window.dart:62]: This is good for this review, but eventually the DimensionsProvider should be returned by the EmbeddingStrategy I think, they're slightly different "views" of the same `hostElement`.

**Review by ditman — APPROVED** [review]
LGTM!

File: lib/web_ui/lib/src/engine/window.dart (hunk 1 above)
Comment (mdebbar) [reply lib/web_ui/lib/src/engine/window.dart:62]: Agreed 👍

**Pull request merged**